cirq-google/cirq_google/engine/engine_client.py

```diff
@@ -744,6 +894,56 @@ async def get_job_results_async(
 
     get_job_results = duet.sync(get_job_results_async)
 
+    def run_job_over_stream(
+        self,
+        project_id: str,
+        program_id: str,
+        code: any_pb2.Any,
+        job_id: Optional[str],
+        processor_ids: Sequence[str],
+        run_context: any_pb2.Any,
+        priority: Optional[int] = None,
+        description: Optional[str] = None,
+        labels: Optional[Dict[str, str]] = None,
+    ) -> duet.AwaitableFuture[quantum.QuantumResult]:
```

Comment:
How are failed jobs or errors conveyed through this interface? https://source.corp.google.com/piper///depot/google3/google/cloud/quantum/v1alpha1/engine.proto;rcl=389782031;l=1207

Reply:
`_manage_stream` first receives these responses through the response iterable, which is then sent through the `ResponseDemux` to the `_make_request` task handling their corresponding jobs. Where `response.result` is checked inside `_make_request`, there will be additional clauses for failed jobs and stream errors.

For retryable errors, the appropriate stream request will be set as `current_request` and the loop continues, so that the request is sent over the stream.

Non-retryable errors are raised, which is propagated to the `duet.AwaitableFuture` for this task. The end user awaiting on this future will see the exception.